x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -4431,11 +4431,16 @@ public void testReplaceSortByExpressionsWithStats() {
      *     \_EsRelation[test][_meta_field{f}#13, emp_no{f}#7, first_name{f}#8, ge..]
      */
     public void testInlinestatsNestedExpressionsInGroups() {
-        var plan = optimizedPlan("""
+        var query = """
             FROM test
             | INLINESTATS c = COUNT(salary) by emp_no % 2
-            """);
-
+            """;
+        if (Build.current().isSnapshot() == false) {
+            var e = expectThrows(ParsingException.class, () -> analyze(query));
+            assertThat(e.getMessage(), containsString("line 2:3: mismatched input 'INLINESTATS' expecting {"));
```

Comment:
`inlinestats` is special in that it's behind a feature_flag, which is not available/used in the parser checks.